### PR TITLE
added lookup, verification of unit type, authreqs for ebevis

### PR DESCRIPTION
**Pull request by erlendoksvoll**

### CODE REVIEW

File: src/Altinn.Dan.Plugin.Arbeidstilsynet/Utils/BRUtils.cs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+namespace Altinn.Dan.Plugin.Arbeidstilsynet.Utils
+{
+    public class BRUtils
+    {
+        public static async Task<BREntityRegisterEntry> GetMainUnit(string organizationNumber, HttpClient client)
+        {
+            var org = await GetOrganizationInfoFromBR(organizationNumber, client);
+
+            if (org != null && !string.IsNullOrEmpty(org.OverordnetEnhet))
```
Comment (ctrlaltd3v) [inline src/Altinn.Dan.Plugin.Arbeidstilsynet/Utils/BRUtils.cs:19]: I know we disagree about this, but: https://programmerbyday.wordpress.com/2020/11/23/why-you-should-always-use-curly-braces-for-single-statement-blocks-in-csharp/

File: src/Altinn.Dan.Plugin.Arbeidstilsynet/Utils/BRUtils.cs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+                return await GetMainUnit(org.OverordnetEnhet.ToString(), client);
```
Comment (ctrlaltd3v) [inline src/Altinn.Dan.Plugin.Arbeidstilsynet/Utils/BRUtils.cs:20]: Redundant ToString(), and the two else's.

File: src/Altinn.Dan.Plugin.Arbeidstilsynet/Utils/BRUtils.cs (hunk 1 above)
Comment (erlendoksvoll) [reply src/Altinn.Dan.Plugin.Arbeidstilsynet/Utils/BRUtils.cs:19]: Agree to disagree :)

File: src/Altinn.Dan.Plugin.Arbeidstilsynet/Models/Unit.cs [hunk 3]
```diff
@@ -0,0 +1,223 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Newtonsoft.Json;
```
Comment (ctrlaltd3v) [inline src/Altinn.Dan.Plugin.Arbeidstilsynet/Models/Unit.cs:4]: We need to choose; System.Text.Json or Newtonsoft.Json.
If we continue with Newtonsoft, then we should remove unused System.Text-usings.

File: src/Altinn.Dan.Plugin.Arbeidstilsynet/Models/Unit.cs [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+
+namespace Altinn.Dan.Plugin.Arbeidstilsynet.Models.Unit
+{
+    // Root myDeserializedClass = JsonConvert.DeserializeObject<Root>(myJsonResponse);
```
Comment (ctrlaltd3v) [inline src/Altinn.Dan.Plugin.Arbeidstilsynet/Models/Unit.cs:8]: Delete?

File: src/Altinn.Dan.Plugin.Arbeidstilsynet/Models/SubUnit.cs [hunk 5]
```diff
@@ -0,0 +1,209 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Text.Json.Serialization;
+using Newtonsoft.Json;
+
+namespace Altinn.Dan.Plugin.Arbeidstilsynet.Models.Sub
+{
+    // Root myDeserializedClass = JsonSerializer.Deserialize<Root>(myJsonResponse);
```
Comment (ctrlaltd3v) [inline src/Altinn.Dan.Plugin.Arbeidstilsynet/Models/SubUnit.cs:9]: Delete?

File: src/Altinn.Dan.Plugin.Arbeidstilsynet/Utils/BRUtils.cs (hunk 1 above)
Comment (altinnadmin) [reply src/Altinn.Dan.Plugin.Arbeidstilsynet/Utils/BRUtils.cs:19]: @ctrlaltd3v It's a well known fact that @erlendoksvoll is wrong on this subject :)

File: src/Altinn.Dan.Plugin.Arbeidstilsynet/Metadata.cs [hunk 6]
```diff
@@ -23,7 +25,7 @@ public List<EvidenceCode> GetEvidenceCodes()
                     EvidenceCodeName = "Bemanningsforetakregisteret",
                     EvidenceSource = EvidenceSourceMetadata.SOURCE,
                     ServiceContext = "eBevis",
-                    AccessMethod = Nadobe.Common.Models.Enums.EvidenceAccessMethod.Open,
+                    AccessMethod = Nadobe.Common.Models.Enums.EvidenceAccessMethod.Open,
```
Comment (ctrlaltd3v) [inline src/Altinn.Dan.Plugin.Arbeidstilsynet/Metadata.cs:28]: Redundant qualifier. More of these in this class, take a look.

File: src/Altinn.Dan.Plugin.Arbeidstilsynet/Models/Unit.cs (hunk 3 above)
Comment (altinnadmin) [reply src/Altinn.Dan.Plugin.Arbeidstilsynet/Models/Unit.cs:1]: If System.Text.Json covers our needs, why would we add a third-party lib dependency?

Comment (erlendoksvoll) [reply src/Altinn.Dan.Plugin.Arbeidstilsynet/Models/Unit.cs:1]: Newtonsoft is so much better.

Comment (elsand) [reply src/Altinn.Dan.Plugin.Arbeidstilsynet/Models/Unit.cs:1]: It has improved quite a bit in .NET5, but Core is still on 3.1 and Common has it as a dependency. We also have a custom attribute we would have to reimplement, so for now we should just use Newtonsoft.

But we should try to get rid of it when getting Core to .NET6.

File: src/Altinn.Dan.Plugin.Arbeidstilsynet/Program.cs [hunk 7]
```diff
@@ -44,16 +42,6 @@ private static Task Main(string[] args)
                     // Client configured with circuit breaker policies
                     services.AddHttpClient("SafeHttpClient", client => { client.Timeout = new TimeSpan(0, 0, 30); })
                         .AddPolicyHandlerFromRegistry("defaultCircuitBreaker");
-
-                    // Client configured without circuit breaker policies. shorter timeout
-                    services.AddHttpClient("CachedHttpClient", client => { client.Timeout = new TimeSpan(0, 0, 5); });
```
Comment (ctrlaltd3v) [inline src/Altinn.Dan.Plugin.Arbeidstilsynet/Program.cs:49]: Remove? Not in use

File: src/Altinn.Dan.Plugin.Arbeidstilsynet/Utils/BRUtils.cs (hunk 1 above)
Comment (elsand) [reply src/Altinn.Dan.Plugin.Arbeidstilsynet/Utils/BRUtils.cs:16]: I'm with @erlendoksvoll on this one. I've always felt the classical "arguments" against dropping braces presented in that blog (and million others parroting the same) are utterly contrived and never something that actually occurs in the real world, while curly braces around the most trivial `if`s just add visual noise and reduces the amount of actual code you can fit in a screenful ultimately causing more scrolling. And if you don't manage your indentations right the code gets unreadable regardless of braces.

Comment (ctrlaltd3v) [reply src/Altinn.Dan.Plugin.Arbeidstilsynet/Utils/BRUtils.cs:16]: Hear, hear! Two against one 😂

File: src/Altinn.Dan.Plugin.Arbeidstilsynet/Models/Unit.cs (hunk 4 above)
Comment (erlendoksvoll) [reply src/Altinn.Dan.Plugin.Arbeidstilsynet/Models/Unit.cs:6]: Delete

File: src/Altinn.Dan.Plugin.Arbeidstilsynet/Main.cs [hunk 8]
```diff
@@ -22,13 +24,13 @@ public class Main
         private ILogger _logger;
         private readonly HttpClient _client;
         private readonly ApplicationSettings _settings;
```
Comment (ctrlaltd3v) [inline src/Altinn.Dan.Plugin.Arbeidstilsynet/Main.cs:26]:
```suggestion
        private readonly IApplicationSettings _settings;
```

File: src/Altinn.Dan.Plugin.Arbeidstilsynet/Main.cs [hunk 9]
```diff
@@ -22,13 +24,13 @@ public class Main
...
-        private readonly EvidenceSourceMetadata _metadata;
+        private readonly IEvidenceSourceMetadata _metadata;
 
-        public Main(IHttpClientFactory httpClientFactory, IApplicationSettings settings)
+        public Main(IHttpClientFactory httpClientFactory, IApplicationSettings settings, IEvidenceSourceMetadata evidenceSourceMetadata)
         {
             _client = httpClientFactory.CreateClient("SafeHttpClient");
             _settings = (ApplicationSettings)settings;
```
Comment (ctrlaltd3v) [inline src/Altinn.Dan.Plugin.Arbeidstilsynet/Main.cs:32]:
```suggestion
            _settings = settings;
```

**Review by ctrlaltd3v — APPROVED** [review]
⭐️

**Pull request merged**